minijinja/src/compiler/parser.rs

```diff
@@ -1062,7 +1078,11 @@ impl<'a> Parser<'a> {
                         return Ok(rv);
                     }
                     rv.push(ok!(self.parse_stmt()));
-                    expect_token!(self, Token::BlockEnd(..), "end of block");
+
+                    // Blockend may not exist, for example a do tag is not a block but a tag.
+                    if let Some((Token::BlockEnd(..), _)) = ok!(self.stream.current()) {
```

Comment:
I'm unsure if this is the best way to check, or would lead to user confusion.

If they are using `enddo`, it will error anyway.

Reply:
This actually is not necessary if you move the incorrect block end expectation. Fixed it.

Reply:
Ah awesome, thanks!